manager/src/main/java/io/gingersnapproject/search/SearchResult.java

```diff
@@ -0,0 +1,17 @@
+package io.gingersnapproject.search;
+
+import java.util.List;
+
+public interface SearchResult {
+
+   long hitCount();
+
+   boolean hitCountExact();
+
+   List<SearchHit> hits();
```

Comment:
Yes ES (and OpenSearch) does not support that.
We could expose a streamed api using the search after (the next generation of scrolling api), but IMO it would be a stretch.
It would be the same as exposing a streaming API from a non-reactive JDBC driver. Nothing good in my opinion.
I think that it would be easier to expose a streaming API when we switch to Hibernate Search.
For the moment I would keep it as it is.

Reply:
I am not sure I like this approach. When I see this method I assume it is not going to block, but from my understanding it could right? We need to make sure to be non blocking from the start and only be that way. If there is a place that is blocking we need to call that out immediately.

And long term this API needs to be non blocking, so I personally feel it is better to have it that way to begin.